Validation/MuonME0Validation/plugins/MuonME0SegHarvestor.cc

```diff
@@ -0,0 +1,159 @@
+#include "FWCore/Framework/interface/EDAnalyzer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "TTree.h"
+#include "TFile.h"
+#include "TGraphAsymmErrors.h"
+#include "FWCore/Framework/interface/EDAnalyzer.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+
+///Geometry
+#include "Geometry/Records/interface/MuonGeometryRecord.h"
+#include "Geometry/CommonDetUnit/interface/GeomDet.h"
+
+#include "Geometry/GEMGeometry/interface/ME0Geometry.h"
+#include "Geometry/GEMGeometry/interface/ME0EtaPartition.h"
+#include "Geometry/GEMGeometry/interface/ME0EtaPartitionSpecs.h"
+#include "Geometry/CommonTopologies/interface/StripTopology.h"
+
+#include "DQMServices/Core/interface/DQMStore.h"
+
+///Log messages
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+
+#include "Validation/MuonME0Validation/plugins/MuonME0SegHarvestor.h"
+
+
+MuonME0SegHarvestor::MuonME0SegHarvestor(const edm::ParameterSet& ps)
+{
+  dbe_path_ = std::string("MuonME0RecHitsV/ME0SegmentsTask/");
+  outputFile_ = ps.getUntrackedParameter<std::string>("outputFile", "myfile.root");
+}
+
+
+MuonME0SegHarvestor::~MuonME0SegHarvestor()
+{
+}
+
+
+TProfile* MuonME0SegHarvestor::ComputeEff(TH1F* num, TH1F* denum, std::string nameHist)
+{
+  std::string name  = "eff_"+nameHist;
+  std::string title = "Segment Efficiency"+std::string(num->GetTitle());
+  TProfile * efficHist = new TProfile(name.c_str(), title.c_str(),denum->GetXaxis()->GetNbins(), denum->GetXaxis()->GetXmin(),denum->GetXaxis()->GetXmax());
+
+  for (int i=1; i <= denum->GetNbinsX(); i++) {
+
+    double nNum = num->GetBinContent(i);
+    double nDenum = denum->GetBinContent(i);
+    if ( nDenum == 0 || nNum ==0  ) {
+      continue;
+    }
+    if ( nNum > nDenum ) {
+      double temp = nDenum;
+      nDenum = nNum;
+      nNum = temp;
+      std::cout<<"Alert! specific bin's num is bigger than denum "<<i<<" "<<nNum<<" "<<nDenum<<std::endl;
+    }
+    const double effVal = nNum/nDenum;
+    efficHist->SetBinContent(i, effVal);
+    efficHist->SetBinEntries(i,1);
+    efficHist->SetBinError(i,0);
+    const double errLo = TEfficiency::ClopperPearson((int)nDenum,(int)nNum,0.683,false);
+    const double errUp = TEfficiency::ClopperPearson((int)nDenum,(int)nNum,0.683,true);
+    const double errVal = (effVal - errLo > errUp - effVal) ? effVal - errLo : errUp - effVal;
+    efficHist->SetBinError(i, sqrt(effVal * effVal + errVal * errVal));
+  }
+  return efficHist;
+}
+
+
+void MuonME0SegHarvestor::ProcessBooking( DQMStore::IBooker& ibooker, DQMStore::IGetter& ig, std::string nameHist, TH1F* num, TH1F* den )
+{
+    
+  if( num !=nullptr && den !=nullptr ) {
+      
+    TProfile* profile = ComputeEff(num, den, nameHist);
```

Comment:
You need to this TProfile, otherwise it leaks. Source is never used for booking, a clone of it is.
https://github.com/cms-sw/cmssw/blob/CMSSW_9_0_X/DQMServices/Core/src/DQMStore.cc#L1463

Or, ideally, just use proper book method (not the one where you pass a TH1 instance).